Gems/Atom/RHI/Code/Include/Atom/RHI/Resource.h

```diff
@@ -83,6 +100,13 @@ namespace AZ::RHI
 
         //! The version is monotonically incremented any time the backing resource is changed.
         uint32_t m_version = 0;
+
+        //! Cache the resourceViews in order to avoid re-creation
+        //! Since ResourceView has a dependency to Resource this cache holds raw pointers here in order to ensure there
+        //! is no circular dependency between the resource and it's resourceview.
+        mutable AZStd::unordered_map<size_t, ResourceView*> m_resourceViewCache;
```

Comment:
So DeviceResource and Resource contains a resource cache? We should add comments on how this is different from DeviceResource one. To be honest it would be nice to have a diagram on the wiki showing the relationship between resource, deviceresouce, devicjeobject, multideviceobject and various pools. I think I went through that exercise once but need it again to refresh my memory :P